_posts/2023-11-20-unlocking-the-secrets-to-ingestion.md

```diff
@@ -0,0 +1,191 @@
+
+# Host Settings
+
+### JVM Optimization
+
+There are a few optimizations to look into when you are setting up your cluster for OpenSearch. For those who may not know, OpenSearch is a project written in Java and it relies on the JVM (Java Virtual Machine) to run. When the JVM starts up it uses a set amount of memory based on the default configuration (1Gb out of the box). This means that even if you attempt to run OpenSearch on a machine that has 32Gb it will not be able to take full advantage of those resources unless you adjust the settings.
+
+For ingest heavy workloads we recommend setting the JVM size to half of your available memory. There are two ways this can be set:
+
+1. JVM config file (tarball install): `/config/jvm.options`
+
+```bash
+# /config/jvm.options
+
+# Min JVM memory size
+- Xms16g
+
+# Max JVM memory size
+- Xmx16g
+```
+
+1. Environment variable (container based installs): `OPENSEARCH_JAVA_OPTS`
+
+```bash
+OPENSEARCH_JAVA_OPTS="-Xms16g -Xmx16g“
+```
+
+### Translog Flush
+
+A translog flush is when OpenSearch commits the items that have been ingested but not persisted to on-disk Lucene segments. By default this happens every time the translog reaches 512MB. Increasing the translog flush size reduces the frequency of these operations which are resource intensive. Additionally they create larger Lucene segments which merge less frequently saving even more resources for ingestion. To see your current translog statistics you can use the [node stats](https://opensearch.org/docs/latest/api-reference/nodes-apis/nodes-stats/) endpoint like so: `curl -XPOST "<cluster url>/<index name>/_stats/flush?pretty"`
+
+We recommend setting the translog flush size to around 25% of your available Java heap defined earlier. On a node with 16GB of memory we would set our heap to 50% of the available memory which would be 8GB for the heap. Then we would take 25% of the heap to set aside for the translog flushes or 2GB.
+
+```json
+POST index-name/_settings
+{
+	"index" {
+		"translog.flush_threshold_size": "2048MB"
+	}
+}
+```
+
+Small note, the translog is not in memory. We set the translog threshold  to 25% of our memory because when the translog flushes it will transfer the data from disk into memory to be compiled into a Lucene segment to be persisted to disk.
+
+# Sharding and Replication
```

Comment:
```suggestion
# Sharding and replication
```

Reply:
Should this be formatted as a level 2 heading? Level 1 headings are used for main titles.